File: Iceland/100/iceland.bitaflipp.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int N, mx = -1, cur = 0, cnt = 0;
	cin >> N;
	string bit, s;
	vector<int> bits(N);
	for (int i = 0; i < N; ++i) {
		cin >> bits[i];
		cnt += bits[i];
		bits[i] = bits[i] == 1 ? -1 : 1;
	}

	
	for (int i = 0; i < N; ++i) {
		cur = max(bits[i], cur + bits[i]);
    mx = max(mx, cur);
	}
cout << mx + cnt << endl;
return 0;
}

File: Iceland/100/iceland.bitaflipp.py
import sys

N = int(input())
cur = 0
cnt = 0
bits = list(map(int, sys.stdin.readline().split()))
cnt = sum(bits)
for i in range(len(bits)):
    bits[i] = -1 if bits[i] == 1 else 1

mx = -1
for i in range (N):
    cur = max(bits[i], cur + bits[i])
    mx = max(mx, cur)
print(mx + cnt)
